src/document.h

```diff
@@ -186,19 +188,31 @@ struct Document {
             loopv(i, sys->imagelist) {
                 Image &image = *sys->imagelist[i];
                 if (image.trefc) {
-                    fos.Write("I", 1);
+                    imt = sys->defaultimagetype;
+                    fos.Write(imagetypeidentifiers[imt][0], 1);
                     sos.WriteDouble(image.display_scale);
-                    if (image.png_data.empty()) {
+                    if (image.image_type != imt || image.image_data.empty()) {
+                        image.image_type = imt;
                         wxImage im = image.bm_orig.ConvertToImage();
-                        im.SaveFile(fos, wxBITMAP_TYPE_PNG);
+                        im.SaveFile(fos, imt);
+
+                        wxMemoryOutputStream mos;
+                        off_t beforeimage = mos.TellO();
+                        im.SaveFile(mos, imt);
+                        off_t afterimage = mos.TellO();
+                        auto sz = afterimage - beforeimage;
+                        image.image_data.resize(sz);
+                        mos.SeekO(beforeimage);
+                        mos.CopyTo(image.image_data.data(), image.image_data.size());
+                        mos.SeekO(afterimage);
+                        mos.Close();

```

Comment:
Automatic refill of the image data vector in case the image type/format has changed or the vector is empty